src/map/skill.cpp

```diff
@@ -1415,11 +1415,10 @@ int skill_additional_effect(struct block_list* src, struct block_list *bl, uint1
 					// Enchanting Shadow gives a chance to inflict shadow wounds to the enemy.
 					if ((sce = sc->data[SC_SHADOW_WEAPON]) && rnd() % 100 < sce->val2) {
 						if (tsc && tsc->data[SC_SHADOW_SCAR]) {
-							short count = 1 + tsc->data[SC_SHADOW_SCAR]->val1;
+							uint16 count = 1 + tsc->data[SC_SHADOW_SCAR]->val1;
 
 							// Need official stack limit. [Rytech]
-							if (count > 5)
-								count = 5;
+							count = min(5, count);
 
 							sc_start(src, bl, SC_SHADOW_SCAR, 100, count, skill_get_time2(SHC_ENCHANTING_SHADOW, sce->val1));
```

Comment:
The count could be handled directly in this sc_start call

Reply:
I had left it separate so the comment stands out on the TODO for it.